### PR TITLE
refactor(cmake): improve cmake and add new feature

### PR DESCRIPTION
**Pull request by asterwyx**
* remove redundant "dtkwidget" in include directory;
 * improve cmake, modify some variable names, use file() to collect source files for scalability;
 * add full target_link_libraries support.

Log: add full target_link_libraries support
Influence: all projects who include headers directly by path

### CODE REVIEW

**Comment by xzl01** [comment]
2023!

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,81 +1,156 @@
...
-set (DVERSION "5.6.2" CACHE STRING "define project version")
-project (DtkWidget
-  VERSION ${DVERSION}
-  DESCRIPTION "DTK Widget module"
-  HOMEPAGE_URL "https://github.com/linuxdeepin/dtkwidget"
-  LANGUAGES CXX C
+cmake_minimum_required(VERSION 3.10)
+
+set(VERSION
+    "5.6.2"
```
Comment (asterwyx) [inline CMakeLists.txt:4]: 改成changelog对应的版本

File: plugin/dtkuidemo/CMakeLists.txt [hunk 2]
```diff
@@ -1,45 +1,22 @@
...
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-
 set(CMAKE_AUTOUIC ON)
-set(CMAKE_AUTOMOC ON)
-set(CMAKE_AUTORCC ON)
-
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core)
```
Comment (asterwyx) [inline plugin/dtkuidemo/CMakeLists.txt:5]: 这里可以去掉

File: plugin/dtkuiplugin/CMakeLists.txt [hunk 3]
```diff
@@ -1,68 +1,47 @@
...
-
-set(CMAKE_AUTOUIC ON)
-set(CMAKE_AUTOMOC ON)
-set(CMAKE_AUTORCC ON)
-
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-
-find_package(DtkGui REQUIRED)
 find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets)
```
Comment (asterwyx) [inline plugin/dtkuiplugin/CMakeLists.txt:6]: 这里可以去掉

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -1,200 +1,173 @@
...
+
+target_include_directories(${BIN_NAME} PRIVATE
+    ${PROJECT_SOURCE_DIR}/src/widgets
+    ${PROJECT_SOURCE_DIR}/include/DWidget
+    ${PROJECT_SOURCE_DIR}/include/util
+    ${PROJECT_SOURCE_DIR}/include/widgets
+    ${PROJECT_SOURCE_DIR}/include/global
+    ${PROJECT_SOURCE_DIR}/include
+    ${CONFIG_INCLUDE}
 )
```
Comment (asterwyx) [inline tests/CMakeLists.txt:171]: 移动到target_link前面

File: CMakeLists.txt [hunk 5]
```diff
@@ -1,81 +1,156 @@
...
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+if(UNIX AND NOT APPLE)
+    set(LINUX TRUE)
+endif()
+
 # Set build option
-set (BUILD_PLUGINS ON CACHE BOOL "BUILD PLUGIN EXAMPLES")
+set(BUILD_PLUGINS
+    ON
+    CACHE BOOL "Build dtkuiplugin example"
```
[the remark this answers is not in the document]
Comment (asterwyx) [reply CMakeLists.txt:33]: Done.

File: CMakeLists.txt (hunk 1 above)
Comment (asterwyx) [reply CMakeLists.txt:4]: Done.

File: debian/rules [hunk 6]
```diff
@@ -20,7 +20,7 @@ endif
 	dh $@ --parallel
```
[the remark this answers is not in the document]
Comment (asterwyx) [reply debian/rules:20]: Done.

File: plugin/dtkuidemo/CMakeLists.txt (hunk 2 above)
Comment (asterwyx) [reply plugin/dtkuidemo/CMakeLists.txt:5]: Done.

File: plugin/dtkuiplugin/CMakeLists.txt (hunk 3 above)
Comment (asterwyx) [reply plugin/dtkuiplugin/CMakeLists.txt:6]: Done.

File: tests/CMakeLists.txt (hunk 4 above)
Comment (asterwyx) [reply tests/CMakeLists.txt:171]: Done.

**Comment by asterwyx** [comment]
> 2023!

Done.

**Review by kegechen — APPROVED** [review]

**Review by Groveer — DISMISSED** [review]

**Review by Groveer — APPROVED** [review]

**Pull request merged**